department-of-veterans-affairs/caseflow
# J hoang/appeals 29213

Resolves [29213](https://jira.devops.va.gov/browse/APPEALS-29213?workflowName=APPEAL+Story+Workflow&stepId=20)

# Description
Added a modal to the Correspondence Review Package page for a user to request the removal of the package.
After requesting for a removal, a RemovePackageTask will be created as a child of the ReviewPackageTask and will be assigned to the MailTeamSupervisor organization.

## Acceptance Criteria

## Testing Plan
1. Login as MAIL_TEAM_SUPERVISOR_ADMIN_USER or JOLLY_POSTMAN
2. Go to `/queue/correspondence`
3. Click on any veteran
4. Click on the "Request package action" from the dropdown
5. Click on "Remove package from Caseflow"
6. Fill in the textbox
7. Click on the Confirm request button
8. Ensure user is taken back to `/queue/correspondence`
9. ![Screenshot 2023-12-12 at 11 22 59 AM](https://github.com/department-of-veterans-affairs/caseflow/assets/111081469/c4acdcad-8cc8-4ed1-8449-c5f86b8bd1ee)



# Frontend
## User Facing Changes

![Screenshot 2023-12-12 at 11 22 25 AM](https://github.com/department-of-veterans-affairs/caseflow/assets/111081469/e1999c34-d660-4dde-9934-2587b62ea067)
![Screenshot 2023-12-12 at 11 22 38 AM](https://github.com/department-of-veterans-affairs/caseflow/assets/111081469/b3860b71-b9ca-4b16-bf32-dc31af55c9ee)

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
